source/common/router/req_header_formatter.cc

```diff
@@ -0,0 +1,73 @@
+#include "common/router/req_header_formatter.h"
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "common/config/rds_json.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+
+namespace Envoy {
+namespace Router {
+
+HeaderFormatterPtr RequestHeaderParser::parseInternal(const std::string& format) {
+  if (format.find("%") == 0) {
+    const size_t last_occ_pos = format.rfind("%");
+    if (last_occ_pos == std::string::npos || last_occ_pos <= 1) {
+      throw EnvoyException(fmt::format("Incorrect header configuration: {}. Expected the variable "
```

Comment:
I'd also add.. `expected "A" but given "B"` to make this message more descriptive.